tests/bdd/features/ha/node-agent/test_path_replacement.py

```diff
@@ -0,0 +1,210 @@
+from openapi.model.volume_policy import VolumePolicy
+from openapi.model.protocol import Protocol
+from openapi.exceptions import ApiException
+
+import cluster_agent_pb2 as pb
+
+VOLUME_UUID = "5cd5378e-3f05-47f1-a830-a0f5873a1449"
+VOLUME_SIZE = 10485761
+POOL_UUID = "4cc6ee64-7232-497d-a26f-38284a444980"
+POOL_NODE = "io-engine-3"
+TARGET_NODE_1 = "io-engine-1"
+TARGET_NODE_2 = "io-engine-2"
+NEXUS_NQN = "nqn.2019-05.io.openebs:%s" % VOLUME_UUID
+# Interval to wait before HA Node agent classifies broken path as failed
+PATH_DETECTION_TIME = 6
+# FIO should be active long enough to outlive the detection interval.
+FIO_RUNTIME = PATH_DETECTION_TIME * 2
+
+
+@scenario(
+    "node_agent.feature",
+    "replace failed I/O path on demand for NVMe controller",
+)
+def test_replace_failed_io_path_on_demand_for_nvme_controller():
+    """replace failed I/O path on demand for NVMe controller."""
+
+
+@given("a client connected to one nexus via single I/O path")
+def a_client_connected_to_one_nexus_via_single_io_path(connect_to_first_path):
+    """a client connected to one nexus via single I/O path."""
+    pass
+
+
+@given(
+    "a control plane, 2 ANA-enabled Io-Engine instances, 1 ANA-enabled host and a published volume"
+)
+def a_control_plane_2_anaenabled_io_engine_instances_1_anaenabled_host_and_a_published_volume(
+    background,
+):
+    """a control plane, 2 ANA-enabled Io-Engine instances, 1 ANA-enabled host and a published volume."""
+    volume = background
+    assert hasattr(volume.state, "target")
+    pass
+
+
+@given("fio client is running against target nexus")
+def fio_client_is_running_against_target_nexus(run_fio_to_first_path):
+    """fio client is running against target nexus."""
+    pass
+
+
+@given("a running ha node agent", target_fixture="ha_node_instance")
+def a_running_ha_node_agent():
+    return HaNodeHandle("127.0.0.1:11600")
+
+
+@when("the only I/O path degrades")
+def the_only_io_path_degrades(degrade_first_path):
+    """the only I/O path degrades."""
+    pass
+
+
+@then("fio client should successfully complete with the replaced I/O path")
+def fio_client_should_successfully_complete_with_the_replaced_io_path(
+    fio_completes_successfully,
+):
+    """fio client should successfully complete with the replaced I/O path."""
+    pass
+
+
+@then("it should be possible to create a second nexus and replace failed path with it")
+def it_should_be_possible_to_create_a_second_nexus_and_replace_failed_path_with_it(
+    publish_to_node_2, replace_failed_path_with_node2
+):
+    """it should be possible to create a second nexus and connect it as the second path."""
+    pass
+
+
+"""" FixTure Implementations """
+
+
+@pytest.fixture
+def background():
+    Deployer.start(
+        3,
+        node_agent=True,
+        cluster_agent=True,
+        cache_period="1s",
+        io_engine_env="NEXUS_NVMF_ANA_ENABLE=1,NEXUS_NVMF_RESV_ENABLE=1",
+        agents_env="TEST_NEXUS_NVMF_ANA_ENABLE=1",
```

Comment:
Probably this will have to stop being a test env and be enabled by default, though we'll need a task to manage the reservations. Is there one for that already?

Reply:
Nope, I presume.
Might want to change it on Mayastor side once we start doing changes there.